pkg/notification/response.go

```diff
@@ -0,0 +1,56 @@
+package notification
+
+import (
+	"time"
+
+	"github.com/aquasecurity/go-version/pkg/semver"
+)
+
+// flexibleTime is a custom time type that can handle
+// different date formats in JSON. It implements the
+// UnmarshalJSON method to parse the date string into a time.Time object.
+type flexibleTime struct {
+	time.Time
+}
+
+type versionInfo struct {
+	LatestVersion string       `json:"latest_version"`
+	LatestDate    flexibleTime `json:"latest_date"`
+}
+
+type announcement struct {
+	FromDate     *time.Time `json:"from_date"`
+	ToDate       *time.Time `json:"to_date"`
+	FromVersion  *string    `json:"from_version"`
+	ToVersion    *string    `json:"to_version"`
+	Announcement string     `json:"announcement"`
+}
+
+type updateResponse struct {
+	Trivy         versionInfo    `json:"trivy"`
+	Announcements []announcement `json:"announcements"`
+	Warnings      []string       `json:"warnings"`
+}
+
+// shoudDisplay checks if the announcement should be displayed
+// based on the current time and version. If version and date constraints are provided
+// they are checked against the current time and version.
+func (a *announcement) shouldDisplay(now time.Time, currentVersion semver.Version) bool {
+	if a.FromDate != nil && now.Before(*a.FromDate) {
```

Comment:
nit: We may be able to check for the zero value by using [IsZero()](https://pkg.go.dev/time#Time.IsZero) instead of making time.Time a pointer?